crates/ruff_linter/src/rules/flake8_use_pathlib/helpers.rs

```diff
@@ -213,3 +217,10 @@ pub(crate) fn is_argument_non_default(arguments: &Arguments, name: &str, positio
 pub(crate) fn is_top_level_expression_call(checker: &Checker) -> bool {
     checker.semantic().current_expression_parent().is_none()
 }
+
+pub(crate) fn is_statement(checker: &Checker) -> bool {
+    matches!(
+        checker.semantic().current_statement(),
+        Stmt::If(_) | Stmt::For(_) | Stmt::While(_)
+    )
```

Comment:
I think this is too narrow a check. Even a simple assignment statement seems like it should cause an unsafe fix because we can't (or at least don't) verify what the user does with the variable:

```py
x = os.rename("a", "b")
```

I think what we actually want to check is that the [`current_statement_parent`](https://github.com/astral-sh/ruff/blob/1134e85972afb77f410eed90c4a86f18ab568b2f/crates/ruff_python_semantic/src/model.rs#L1261) is  a `Stmt::Expr`:


```suggestion
    checker.semantic().current_statement.is_expr_stmt()
```

Note that this negates the condition, so I guess a better name for the function would be `is_top_level_statement`, and the call sites will need a `!`.

Applying this locally caused tests for a few additional rules to fail, which supports my concern in the other comment.